### PR TITLE
[RFR] Add support for defaultValue and actions in <Create> and <Edit> components

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Add support for `actions` in `<Create>` and `<Edit>` components
- [x] Add support for `defaultValue` in `<Create>` and `<Edit>` components
- [x] Add support for `defaultValue` in `<Input>` components
- [x] Add Refresh button in `<Edit>` view
- [x] Add double submission protection in `<SaveButton>`
- [x] Add animation on click in `<SaveButton>`
- [x] Add doc

## Actions

You can replace the list of default actions by your own element using the `actions` prop:

```js
import { CardActions } from 'material-ui/Card';
import FlatButton from 'material-ui/FlatButton';
import NavigationRefresh from 'material-ui/svg-icons/navigation/refresh';
import { ListButton, ShowButton, DeleteButton } from 'admin-on-rest/lib/mui';

const cardActionStyle = {
    zIndex: 2,
    display: 'inline-block',
    float: 'right',
};

const PostEditActions = ({ basePath, data, refresh }) => (
    <CardActions style={cardActionStyle}>
        <ShowButton basePath={basePath} record={data} />
        <ListButton basePath={basePath} />
        <DeleteButton basePath={basePath} record={data} />
        <FlatButton primary label="Refresh" onClick={refresh} icon={<NavigationRefresh />} />
        {/* Add your custom actions */}
        <FlatButton primary label="Custom Action" onClick={customAction} />
    </CardActions>
);

export const PostEdit = (props) => (
    <Edit actions={<PostEditActions />} {...props}>
        ...
    </Edit>
);
```

## Default Values

To define default values, you can add a `defaultValue` prop to both form components (`<Edit>` and `<Create>`), or add a `defaultValue` from to individual input components.

### Global Default Value

The value of the form `defaultValue` prop must be an object, specifying default value for the created record. For instance:

```js
const postDefaultValue = { created_at: new Date(), nb_views: 0 };
export const PostCreate = (props) => (
    <Create {...props} defaultValue={postDefaultValue}>
        <TextInput source="title" />
        <RichTextInput source="body" />
        <NumberInput source="nb_views" />
    </Create>
);
```

**Tip**: You can include properties in the form `defaultValue` that are not listed as input components, like the `created_at` property in the previous example.

### Per Field Default Value

Alternatively, you can specify a `defaultValue` prop directly in `<Input>` components. Admin-on-rest will merge the child default values with the form default value (input > form):

```js
export const PostCreate = (props) => (
    <Create {...props}>
        <TextInput source="title" />
        <RichTextInput source="body" />
        <NumberInput source="nb_views" defaultValue={0} />
    </Create>
);
```

Supersedes #178

### CODE REVIEW

File: docs/CreateEdit.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+export const PostEdit = (props) => (
+    <Edit actions={<PostEditActions />} {...props}>
+        ...
+    </Edit>
+);
+```
+
+## Default Values
+
+To define default values, you can add a `defaultValue` prop to both form components (`<Edit>` and `<Create>`), or add a `defaultValue` from to individual input components.
```
Comment (jpetitcolas) [inline docs/CreateEdit.md]: s/from to/to/

File: example/posts.js [hunk 2]
```diff
@@ -72,9 +72,9 @@ export const PostCreate = (props) => (
         <TextInput label="Password (if protected post)" source="password" type="password" />
         <TextInput source="teaser" options={{ multiLine: true }} />
         <RichTextInput source="body" />
-        <DateInput label="Publication date" source="published_at" />
+        <DateInput label="Publication date" source="published_at" defaultValue={new Date()}  />
```
Comment (djhi) [inline example/posts.js]: s/{new Date()}  /{new Date()} /

Comment (djhi) [inline example/posts.js]: double spaces

**Comment by fzaninotto** [comment]
Review applied

**Pull request merged**